### PR TITLE
Add support for github enterprise

### PR DESCRIPTION
**Pull request by danmanirl**
If using github enterprise, the hostname will be different to `github.com`

This change adds support for changing the url to something else.

### CODE REVIEW

**Comment by alexduf** [comment]
Hi @danmanirl thanks for the PR.
Unfortunately I can't use gnome anymore, have you had a chance to test that change?

**Comment by danmanirl** [comment]
Hi Alex, 
yes. It's been running in my system for the past day and seems to work, I get the number beside the icon and clicking on it brings me to the notifications page:
Screenshot from 2019-11-29 16-12-33

Should I also get a message in the message tray? The code seems to suggest that I should but I'm not getting any.  I haven't worked with GNOME application development before so any tips for debugging would be appreciated.

**Comment by alexduf** [comment]
>Should I also get a message in the message tray? The code seems to suggest that I should but I'm not getting any

No there's no support for that feature although it's been requested a few times.

I'll try to get more testers, before it gets merged. Again really appreciate the effort I'll try my best to avoid you waiting too long.

**Comment by danmanirl** [comment]
> No there's no support for that feature although it's been requested a few times.

This is the line that I thought was sending a message to the message tray: https://github.com/alexduf/gnome-github-notifications/blob/master/extension.js#L262 . What does it do?

> I'll try to get more testers, before it gets merged. Again really appreciate the effort I'll try my best to avoid you waiting too long.

Great thanks. No rush, thanks for your work.

**Comment by alexduf** [comment]
My bad, I totally forgot about that feature!
It does indeed send a notification to the desktop, it was implemented by @Shelvak 

https://github.com/alexduf/gnome-github-notifications/pull/16

**Comment by Shelvak** [comment]
Hi guys, yes the feature worked when I implemented it, but in the last version[s] of gnome stop working... I have to debug why... (I noticed the bug a few days ago haha).

The review looks good to me, I can patch my local extension and test this week.

**Comment by Shelvak** [comment]
Hey @danmanirl I just tested in my pc and I have to change all the `'https://'baseUrl'/notifications'` for `'https://' + self.baseUrl + '/notifications'`. 
This really worked for you?

Other comment about the review is to change the prop `baseUrl` for `domain` =) If I detect anything else I'll comment here.

Cheers

**Comment by danmanirl** [comment]
> This really worked for you?

It worked. Got the number beside the icon and clicking on it went to the correct page...  Not a javascript programmer though so don't really know why it would/wouldn't work. I'll switch to your suggestion.

File: schemas/org.gnome.shell.extensions.github.notifications.gschema.xml [hunk 1]
```diff
@@ -5,6 +5,11 @@
       <default>''</default>
       <summary>Your github handle</summary>
     </key>
+    <key name="url" type="s">
```
Comment (Shelvak) [inline schemas/org.gnome.shell.extensions.github.notifications.gschema.xml]:
```suggestion
    <key name="domain" type="s">
```

File: prefs.js [hunk 2]
```diff
@@ -78,6 +80,14 @@ function buildPrefsWidget() {
   tokenBox.pack_end(tokenEntry, true, true, 5);
   box.pack_start(tokenBox, false, false, 5);
 
+  const urlBox = new Gtk.Box({orientation: Gtk.Orientation.HORIZONTAL, spacing: 5});
+  const urlLabel = new Gtk.Label ({label : "Github Hostname"});
+  urlBox.pack_start(urlLabel, false, false, 5);
+  const urlEntry = new Gtk.Entry();
+  settings.bind('url', urlEntry, 'text', Gio.SettingsBindFlags.DEFAULT);
+  urlBox.pack_end(urlEntry, true, true, 5);
```
Comment (Shelvak) [inline prefs.js]: domain =)

File: extension.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ const GithubNotifications = new Lang.Class({
   },
 
   reloadSettings: function() {
+    this.domain = Settings.get_string('url');
```
Comment (Shelvak) [inline extension.js]:
```suggestion
    this.domain = Settings.get_string('domain');
```

**Comment by Shelvak** [comment]
Haha, fix for the notifications https://github.com/alexduf/gnome-github-notifications/pull/24
Something "important" changed in gnome because in old versions I didn't have to bind the functions :P

File: schemas/org.gnome.shell.extensions.github.notifications.gschema.xml [hunk 4]
```diff
@@ -5,6 +5,11 @@
       <default>''</default>
       <summary>Your github handle</summary>
     </key>
+    <key name="domain" type="s">
+      <default>'github.com'</default>
+      <summary>Your github url</summary>
+      <description>Set this variable if you are using GitHub Enterprise</description>
```
Comment (Shelvak) [inline schemas/org.gnome.shell.extensions.github.notifications.gschema.xml]: `<summary>Your github domain</summary>`
`<description>Change this domain only if you are using GitHub Enterprise</description>`

**Comment by alexduf** [comment]
Is it working for you then @Shelvak ?

**Comment by Shelvak** [comment]
Just one more change  =) After that I can test again and confirm =) 
https://github.com/alexduf/gnome-github-notifications/pull/23/files#r352709387

**Comment by alexduf** [comment]
I'm not sure if you're aware but since the preferences have been updated you'll have to recompile the schema:
https://developer.gnome.org/gio/stable/glib-compile-schemas.html

**Comment by alexduf** [comment]
I've submitted the latest version for review today. Thanks again everyone I appreciate the help!

**Comment by danmanirl** [comment]
Thanks @alexduf  and @Shelvak for very timely reviews and replies.

**Comment by alexduf** [comment]
It's now officially released, thanks again everyone!

**Pull request merged**